kubectl-cloudflow/deploy/deploy.go

```diff
@@ -23,19 +27,9 @@ import (
 // GetCloudflowApplicationDescriptorFromDockerImage pulls a image and extracts the Cloudflow Application descriptor from a docker label
 func GetCloudflowApplicationDescriptorFromDockerImage(dockerRegistryURL string, dockerRepository string, dockerImagePath string) (domain.CloudflowApplicationSpec, docker.PulledImage) {
 
-	apiversion, apierr := exec.Command("docker", "version", "--format", "'{{.Server.APIVersion}}'").Output()
-	if apierr != nil {
-		util.LogAndExit("Could not get docker API version, is the docker daemon running? API error: %s", apierr.Error())
-	}
-
-	trimmedapiversion := strings.Trim(string(apiversion), "\t \n\r'")
-	client, error := docker.GetClient(trimmedapiversion)
+	client, error := docker.GetVersionedClient()
 	if error != nil {
-		client, error = docker.GetClient("1.39")
-		if error != nil {
-			fmt.Printf("No compatible version of the Docker server API found, tried version %s and 1.39", trimmedapiversion)
-			panic(error)
-		}
+		panic(error)
```

Comment:
Can we not panic here (https://github.com/golang/go/wiki/CodeReviewComments#dont-panic)?

Reply:
Replaced this with `LogAndExit` but did not change in all places since I am not sure about the rationale of the original author. May take it on a case by case basis

Reply:
I think this is a great one for code style / rules. I think we should almost never panic in  the CLI. We need to always provide the user with simple to understand feedback.